docs/source/hyper-parameter-search.rst

```diff
@@ -1,33 +1,232 @@
+   from sklearn.linear_model import SGDClasifier
+
+   df = pd.read_parquet("data/0.parquet")  # data to train on; 23MB as above
+
+   model = SGDClasifier()
+
+   # not compute constrained
+   params = {"l1_ratio": uniform(0, 1)}
+
+   # compute constrained
+   params = {
+       "l1_ratio": uniform(0, 1),
+       "alpha": loguniform(1e-5, 1e-1),
+       "penalty": ["l2", "l1", "elasticnet"],
+       "learning_rate": ["invscaling", "adaptive"],
+       "power_t": uniform(0, 1),
+       "average": [True, False],
+   }
+
+These issues are independent and both can happen the same time. Dask-ML has
+tools to address all 4 combinations. Let's look at each case.
+
+Neither compute nor memory constrained
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This case happens when there aren't many hyperparameters to tune and the data
+fits in memory. This is common when the search doesn't take too long to run.
+
+Scikit-learn can handle this case:
+
+.. autosummary::
+   sklearn.model_selection.GridSearchCV
+   sklearn.model_selection.RandomizedSearchCV
+
+Dask-ML also has some drop in replacements for the Scikit-learn versions that
+works well with Dask.
+
+.. autosummary::
+   dask_ml.model_selection.GridSearchCV
+   dask_ml.model_selection.RandomizedSearchCV
+
+
+These models work well with Dask Arrays/DataFrames. By default, these
+estimators will pass the entire dataset to ``fit`` even if a Dask
+Array/DataFrame is passed.  More detail is in
+":ref:`works-with-dask-collections`".
+
+These estimators make some progress into hyperparameter searches that are
+"compute constrained" or "memory constrained". Details on the cases are
+mentioned below in ":ref:`hyperparameter.mem-ncpu`"and
+":ref:`hyperparameter.cpu-nmem`".
+
+.. _hyperparameter.mem-ncpu:
+
+Memory constrained, but not compute constrained
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This case happens when the data doesn't fit in memory but there aren't many
+hyperparameters to search over. The data doesn't fit in memory, so it makes
+sense to call ``partial_fit`` on each chunk of a Dask Array/Dataframe. This
+estimators does that:
+
+.. autosummary::
+   dask_ml.model_selection.IncrementalSearchCV
+
+More detail on :class:`~dask_ml.model_selection.IncrementalSearchCV` is in
+":ref:`hyperparameter.incremental`".
+
+:class:`~dask_ml.model_selection.GridSearchCV` and
+:class:`~dask_ml.model_selection.RandomizedSearchCV` can to also call
+``partial_fit`` on each chunk of a Dask array, as long as the model passed is
+wrapped with :class:`~dask_ml.wrappers.Incremental`.
+
+.. _hyperparameter.cpu-nmem:
+
+Compute constrained, but not memory constrained
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This case happens when the data fits on in the memory of one machine but when
+there are a lot of hyperparameters to search. The best class for this case is
+:class:`~dask_ml.model_selection.HyperbandSearchCV`:
+
+.. autosummary::
+   dask_ml.model_selection.HyperbandSearchCV
```

Comment:
Does including this here imply that it doesn't work with compute *and* memory constrained? Or should I not read it that way?

Reply:
> it doesn't work with compute and memory constrained?

No. Hyperband the like reduce the amount of computation required, regardless of data size. That is, they fit in "compute constrained" regardless of "memory constrained" status. I've added a note in "compute and memory constrained":

> **Compute and memory constrained**
> * HyperbandSearchCV
> * ...
>
>  These classes work well with data that does not fit in memory. They also reduce the computation required as described in "Compute constrained but not memory constrained"

Does that clear your confusion?